pdfrw/objects/pdfstring.py

```diff
@@ -316,6 +329,11 @@ class PdfString(str):
     indirect = False
 
 
+    # The byte order mark, in both bytes and unicode form
+
+    bytes_bom = codecs.BOM_UTF16_BE
+    uni_bom = bytes_bom.decode('latin-1')
```

Comment:
This is tricky, because Adobe's use of the BOM may be subtly different than the canonical use.

Let's start with the decoder.  The decoder looks at the first two bytes to make a decision.  As Adobe says, "This mechanism precludes beginning a string using PDFDocEncoding with the two characters `thorn` `ydieresis`, which is unlikely to be a meaningful beginning of a word or phrase."

When the decoder sees those two bytes, it decodes using UTF-16, but, most importantly, it _discards the two bytes before doing the decode_, which means that for full transparency and
ability to round-trip properly, the encoder has to add those two bytes when encoding a UTF-16 string.

But, just as importantly, the encoder needs to insure that those two bytes are not present at the start of a string that is _not_ encoded with UTF-16.

So, if the source Unicode starts with `U+FEFF`, you are absolutely right that the encoder must use UTF-16.  OTOH, the encoder must use UTF-16 if that character code (or any other character code not supported by pdfdocencoding) appears _anywhere_ in the string.  This is enforced by the exception raised by `encode('pdfdocencoding')`. 

But `thorn` and `ydieresis` are legal characters in a PDFDocEncoding string, except together right at the start.  So if a source string starts with these two characters, then for the string to be properly encoded, an additional FEFF must be prepended, and it must be encoded in UTF-16 mode.

That check is what this constant is used for.

Having said all that, since it _is_ very confusing, maybe having BOM in the name is incorrect and part of the problem.  We could rename the constant to something like `escape_check` or anything else that might make sense to you...

FWIW, I have added tests to insure that the round-trip is happening properly.  If you change this constant (and fwiw, Python won't let you change it to the real Unicode BOM, but to check that the test does something useful, you could change it to something like u'your bom here'), you will see that it creates test failures in test_pdfstring.py.